pkg/onchain/sc.go

```diff
@@ -22,20 +22,21 @@ type OperationWithEventResponse struct {
 	OperationResponse sendOperation.OperationResponse
 }
 
-/*
-This function send a callSC request to the node.
-After a successful execution, listening to the events emitted by the operation
-should be done on front end side by the consumer.
-However in the current state of things the easiest way to unblock us is to
-listen to these events in MassaStation and return them as a response.
-Hence this function also listen for the first event emitted in an OP and returns it.
-*/
+// CallFunction calls a function of a smart contract on the blockchain. It returns the event generated
+// by the smart contract and an Error.
+// If `async` is true, it returns the operation response and does not wait for the event.
+// Otherwise, it waits for the first event generated by the smart contract and returns it.
+//
+// Note: Listening to the first event generated by the smart contract should be done by the front end,
+// but for now, it is done here to match the previous implementation.
```

Comment:
I'm not sure about that...

Reply:
yes, but now that we implement parameter async bool, does this comment still make sense?

Reply:
Hum yes you are right, we could remove it